plugins/main/macvlan/macvlan.go

```diff
@@ -382,13 +382,13 @@ func cmdAdd(args *skel.CmdArgs) error {
 }
 
 func cmdDel(args *skel.CmdArgs) error {
-	n, _, err := loadConf(args, args.Args)
-	if err != nil {
-		return err
+	var n NetConf
```

Comment:
I guest you needed the external err variable because the rest of the code is assuming there is one already declared, right?

You could simply have 
err := json.Unmarshal(args.StdinData, &n)

and then check the error variable. err will be created and used by the rest of the code.

Reply:
Actually, I want to write in this way, We just add one line:

```
+       var err error
	if err = json.Unmarshal(args.StdinData, &n); err != nil {
		return fmt.Errorf("failed to unmarshal net conf: %v", err)
        }
       ....
```

If write in your way:

```
    	err := json.Unmarshal(args.StdinData, &n)
+       if err != nil {
		return fmt.Errorf("failed to unmarshal net conf: %v", err)
        }
```

Both are ok for me.